examples/kv-events/offline/publisher.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2025 The llm-d Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"sync/atomic"
+
+	zmq "github.com/pebbe/zmq4"
+	"github.com/vmihailenco/msgpack/v5"
+	"k8s.io/klog/v2"
+)
+
+// Publisher sends KV cache events to a ZMQ endpoint.
+type Publisher struct {
+	socket   *zmq.Socket
+	endpoint string
+	seqNum   uint64
+}
+
+// NewPublisher creates a new ZMQ publisher.
+// endpoint is the ZMQ address to bind to (e.g., "tcp://*:5557").
+func NewPublisher(endpoint string) (*Publisher, error) {
+	socket, err := zmq.NewSocket(zmq.PUB)
```

Comment:
from extensibility perspective, you may want to wrap the actual channel behind a wrapper interface.
Optional implementations; ZMQ, HTTP SSE, NATS, test mocks, ...
This could let you reuse more of the code in testing (eg `NewPublished(ch kv-events.Channel`)

Reply:
Delegating to a followup. Opened issue.